src/sentry_value.c

```diff
@@ -1124,6 +1124,44 @@ sentry_value_new_stacktrace(void **ips, size_t len)
     return stacktrace;
 }
 
+sentry_value_t
+sentry_value_new_transaction_context(const char *name)
+{
+    sentry_value_t transaction_context = sentry_value_new_object();
+
+    sentry_transaction_context_set_name(transaction_context, name);
+
+    return transaction_context;
+}
+
+void
+sentry_transaction_context_set_name(
+    sentry_value_t transaction_context, const char *name)
+{
+    sentry_value_t sv_name = sentry_value_new_string(name);
+    if (sentry__string_eq(name, "") || sentry_value_is_null(sv_name)) {
+        sentry_value_decref(sv_name);
+        sv_name = sentry_value_new_string("<unlabeled transaction>");
```

Comment:
As we discussed elsewhere, not sure if we should do validation here. But I don’t object to it either.

Reply:
i'll leave a note for the time being so the validation is still being done somewhere (in case it gets forgotten). the only other place where this validation makes sense to occur is yet to show up in a proper PR, so i think we can revisit this conversation once the implementation for that appears.